Makefile

```diff
@@ -15,9 +15,13 @@ GIT_COMMIT := $(shell git rev-list -1 HEAD | cut -c 1-6)
 SOURCES := $(shell find ./ -name '*.go')
 PWD=$(shell pwd)
 
+FMT=$(fmt)
+UI=$(ui)
+UI_DEV=$(ui-dev)
+
 all: build
 
-build: $(SOURCES)
+build: $(UI) $(FMT) $(SOURCES)
```

Comment:
This makes go fmt output stuff every time you build. The fmt command above doesn't actually apply the format, mainly because I don't agree with everything it formats (I prefer var % 13 over var%13, stupid but I don't like the squished spacing). Adding $(UI) also forces the UI to rebuild each time as well, even if no changes are made to the UI.

Reply:
ok, fmt can be removed, but running gofmt can work.
my understanding was that the dependent builds do not run if they aren't up to date - meaning it'll run UI FMT SOURCES if they are out of sync.

this is a requirement because the go sources rely on "build" from go embed. if there's a more elegant way to do it I'm all ears

Reply:
That's my understanding of make behavior as well, but maybe make isn't seeing something as up to date? I put in your change, then would make a .go code change, 'make' to rebuild, and it would rebuild the UI every time.